runhouse/resources/hardware/cluster.py

```diff
@@ -2107,10 +2167,15 @@ def remove_conda_env(
 
     def download_cert(self):
         """Download certificate from the cluster (Note: user must have access to the cluster)"""
-        self.call_client_method("get_certificate")
-        logger.info(
-            f"Latest TLS certificate for {self.name} saved to local path: {self.cert_config.cert_path}"
-        )
+        try:
+            self.call_client_method("get_certificate")
+            logger.info(
+                f"Latest TLS certificate for {self.name} saved to local path: {self.cert_config.cert_path}"
+            )
+        except ConnectionError:
```

Comment:
whats the rationale here for logging the error instead raising an exception in the client?

Reply:
I thought it will be more user friendly to log an error with a relevant msg, but we could also raise an exception instead, something like:
```
except ConnectionError:
    raise ValueError/RuntimeError(
        "Failed to update periodic cluster status checks: unable to connect to cluster's HTTP client."
    )
```

Reply:
the exception is already being raised tho inside of `call_client_method`